Kitodo-API/src/main/java/org/kitodo/api/ugh/DocStructInterface.java

```diff
@@ -0,0 +1,628 @@
+     *             case, the meta-data element cannot be added because we cannot
+     *             check whether the the meta-data type is allowed or not.
+     */
+    boolean addMetadata(MetadataInterface metadata) throws MetadataTypeNotAllowedException;
+
+    /**
+     * Adds a meta data field with the given name to this DocStruct and sets it
+     * to the given value.
+     *
+     * @param metadataType
+     *            name of the meta data field to add
+     * @param value
+     *            value to set the field to
+     * @return {@code this}, for method chaining. The return value is never
+     *         used.
+     * @throws MetadataTypeNotAllowedException
+     *             if no corresponding MetadataType object is returned by
+     *             getAddableMetadataTypes()
+     */
+    DocStructInterface addMetadata(String metadataType, String value) throws MetadataTypeNotAllowedException;
+
+    /**
+     * Adds a meta-data group to this instance. The method checks, if it is
+     * allowed to add it, based on the configuration. If so, the object is added
+     * and the method returns {@code true}, otherwise it returns {@code false}.
+     * The {@code MetadataGroup} object must already include all necessary
+     * information, such as {@code MetadataGroupType} and value.
+     *
+     * <p>
+     * For internal reasons, this method replaces the {@code MetadataGroupType}
+     * object by a local copy, which is retrieved from the {@code DocStructType}
+     * of this instance. The internal name of both {@code MetadataGroupType}
+     * objects will still be identical afterwards. If a local copy cannot be
+     * found, which means that the meta-data type is invalid on this instance,
+     * false is returned.
+     *
+     * @param metadataGroup
+     *            meta-data group to be added
+     * @return true, if the meta-data group was successfully added, false
+     *         otherwise
+     * @throws MetadataTypeNotAllowedException
+     *             if the {@code DocStructType} of this instance does not allow
+     *             the {@code MetadataGroupType}, or if the maximum number of
+     *             meta-data groups of this type has already been added
+     * @throws DocStructHasNoTypeException
+     *             if no {@code DocStructType} is set on this instance. In this
+     *             case, the meta-data group cannot be added because we cannot
+     *             check whether the the meta-data group type is allowed or not.
+     */
+    boolean addMetadataGroup(MetadataGroupInterface metadataGroup) throws MetadataTypeNotAllowedException;
+
+    /**
+     * Adds a person to the document structure.
+     *
+     * @param person
+     *            person to add
+     * @return always {@code true}. The return value is never used.
+     */
+    boolean addPerson(PersonInterface person) throws MetadataTypeNotAllowedException;
```

Comment:
1) See my comment to the setLocation method.
2) Please adjust the JavaDoc or method signature as they are different.

Reply:
1. as above, cannot be changed
2. You know my opinion: I think, for this throw-away code, adding Javadoc at all feels already a waste of project time. As far as I know, explaining exceptions in Javadoc is optional, and I would recommend to restrain from doing so if, as in this case, [there is no explaination in legacy UGH](https://github.com/kitodo/kitodo-ugh/blob/master/src/main/java/ugh/dl/DocStruct.java#L2762) that can be fallen back to.

Reply:
Your are contributing code regarding the coding guidelines which you accepted. The rules are in this guidelines and I watching for it that nobody is break the rules.